### PR TITLE
Update swift syntax in code gen

### PR DESCRIPTION
**Pull request by kimdv**

### CODE REVIEW

File: CodeGeneration/Sources/Utils/CodeGenerationFormat.swift [hunk 1]
```diff
@@ -66,7 +66,7 @@ public class CodeGenerationFormat: BasicFormat {
 
   public override func visit(_ node: MemberDeclListItemSyntax) -> MemberDeclListItemSyntax {
     let formatted = super.visit(node)
-    if node.indexInParent != 0 && !node.decl.is(EnumCaseDeclSyntax.self) {
+    if !node.decl.is(EnumCaseDeclSyntax.self) {
```
Comment (kimdv) [inline CodeGeneration/Sources/Utils/CodeGenerationFormat.swift:69]: This got internal. 
So now we get a newline on each start. 

Not sure this is by design or an bug?

**Comment by kimdv** [comment]
@swift-ci please test

File: CodeGeneration/Sources/Utils/CodeGenerationFormat.swift [hunk 2]
```diff
@@ -66,7 +66,7 @@ public class CodeGenerationFormat: BasicFormat {
 
   public override func visit(_ node: MemberDeclListItemSyntax) -> MemberDeclListItemSyntax {
     let formatted = super.visit(node)
-    if node.indexInParent != 0 && !node.decl.is(EnumCaseDeclSyntax.self) {
+    if node.index != node.parent?.children(viewMode: .sourceAccurate).first?.index && !node.decl.is(EnumCaseDeclSyntax.self) {
```
Comment (ahoppen) [inline CodeGeneration/Sources/Utils/CodeGenerationFormat.swift:69]: I don’t actually think you need to compare the indices here, comparing the nodes themselves should be sufficient

```suggestion
    if node != node.parent?.children(viewMode: .sourceAccurate).first? && !node.decl.is(EnumCaseDeclSyntax.self) {
```

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntaxbuilder/BuildableNodesFile.swift [hunk 3]
```diff
@@ -94,13 +91,15 @@ private func createConvenienceInitializer(node: Node) throws -> InitializerDeclS
     return nil
   }
 
-  let params = ParameterClauseSyntax {
-    FunctionParameterSyntax("leadingTrivia: Trivia? = nil", for: .functionParameters)
-    for param in normalParameters + builderParameters {
-      param
+  let params = ParameterClauseSyntax(
+    parameterList: FunctionParameterListSyntax {
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntaxbuilder/BuildableNodesFile.swift:95]: Do you know why we can’t use the result builder syntax like before anymore and now need to write `parameterList: FunctionParameterListSyntax`? Are we missing a convenience initializer now?

**Review by ahoppen — COMMENTED** [review]
Thank you for always keeping CodeGeneration up-to-date.

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntaxbuilder/BuildableNodesFile.swift (hunk 3 above)
Comment (kimdv) [reply CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntaxbuilder/BuildableNodesFile.swift:95]: My fault. Sorry

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test windows

**Comment by kimdv** [comment]
@swift-ci please test windows

**Pull request merged**